sdk/core/Azure.Core.Experimental/src/MutableJsonDocument.WriteTo.cs

```diff
@@ -186,7 +187,16 @@ private void WriteString(string path, int highWaterMark, ref Utf8JsonReader read
                 return;
             }
 
-            writer.WriteStringValue(reader.ValueSpan);
+#if NET6_0_OR_GREATER
+            Span<byte> span = stackalloc byte[reader.ValueSpan.Length + 2];
+            span[0] = (byte)'"';
+            reader.ValueSpan.CopyTo(span.Slice(1));
+            span[span.Length - 1] = (byte)'"';
+            writer.WriteRawValue(span);
```

Comment:
Utf8JsonReader has a [ValueIsEscaped property](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonreader.valueisescaped) we could check prior to these calls, but it's only available in .NET 7+, arg.